cmd/server/main.go

```diff
@@ -1,14 +1,61 @@
 package main
 
 import (
-	"fmt"
+	"context"
+	"errors"
+	"log"
 	"net/http"
+	"os"
+	"os/signal"
+	"syscall"
+	"time"
 )
 
-func main() {
-	err := http.ListenAndServe(":8080", http.FileServer(http.Dir("./assets")))
-	if err != nil {
-		fmt.Println("Failed to start server", err)
+// Opens channel and provides a safe way to exit channel upon termination
+func createChannel() (chan os.Signal, func()) {
+	stopCh := make(chan os.Signal, 1)
+	signal.Notify(stopCh, os.Interrupt, syscall.SIGTERM, syscall.SIGINT)
+
+	return stopCh, func() {
+		close(stopCh)
+	}
+}
+
+// Begins http server functionality
+func start(server *http.Server) {
+	log.Println("Server Started")
+	if err := http.ListenAndServe(":8080", http.FileServer(http.Dir("./assets"))); err != nil && !errors.Is(err, http.ErrServerClosed) {
+		log.Println("Failed to start server", err)
 		return
```

Comment:
problem: 
instead of 
```
log.Println("Failed to start server", err)
return
```

should do `log.Fatalf`. 
If  server is not able to serve and the error is not because it is closed, it should be immediately exited, no need to continue the program(creating channels of signals etc)